internal/pkg/agent/install/uninstall.go

```diff
@@ -233,19 +234,34 @@ func applyDynamics(ctx context.Context, log *logger.Logger, cfg *config.Config)
 	inputs, ok := transpiler.Lookup(ast, "inputs")
 	if ok {
 		varsArray := make([]*transpiler.Vars, 0)
-		var wg sync.WaitGroup
-		wg.Add(1)
+
+		// Gie some time for the providers to replaces the variables
+		const timeout = 15 * time.Second
+		var doOnce sync.Once
+		ctx, cancel := context.WithTimeout(ctx, timeout)
+
```

Comment:
add 
```
defer cancel()
```
there is a possibility that the 
```
ctrl, err := composable.New(log, cfg)
```
returns the error and the ```cancel``` is never called